### PR TITLE
feat: add deploy script with basic auth for stage

### PR DESCRIPTION
**Pull request by AlcibiadesCleinias**

### CODE REVIEW

File: subgraph/fluence-graph.sh [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+# Validate envs.
+if [ "$BASIC_AUTH_FLUENCE_STAGE" = "" ]; then
+  echo "Please provide 'BASIC_AUTH_FLUENCE_STAGE' env."
+  exit 0
+fi
+
+# Choose auth string.
+basic_auth=""
+if [ "$BASIC_AUTH_FLUENCE_STAGE" != "" ]; then
+  basic_auth=${BASIC_AUTH_FLUENCE_STAGE}@
```
Comment (nahsi) [inline subgraph/fluence-graph.sh:22]:
```suggestion
  basic_auth="${BASIC_AUTH_FLUENCE_STAGE}@"
```

File: subgraph/fluence-graph.sh [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+# Validate envs.
+if [ "$BASIC_AUTH_FLUENCE_STAGE" = "" ]; then
+  echo "Please provide 'BASIC_AUTH_FLUENCE_STAGE' env."
+  exit 0
+fi
+
+# Choose auth string.
+basic_auth=""
+if [ "$BASIC_AUTH_FLUENCE_STAGE" != "" ]; then
```
Comment (nahsi) [inline subgraph/fluence-graph.sh:21]:
```suggestion
if [[ -z $BASIC_AUTH_FLUENCE_STAGE ]]; then
```

`[[` is bash builtin command, `[` is legacy command outside of bash, if I'm not mistaken.

It is a minor thing but according to modern bash guidelines it is better to use `[[` :)
http://mywiki.wooledge.org/BashGuide/TestsAndConditionals

`-z` check means `var is undefined or empty`

File: subgraph/fluence-graph.sh [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+# >> export BASIC_AUTH_FLUENCE_STAGE=<check .example.env> && ./fluence-graph.sh
+
+# Args:
+# $1 - action {deploy, create}.
+
+# Envs:
+# - BASIC_AUTH_FLUENCE_STAGE
+
+# Validate envs.
+if [ "$BASIC_AUTH_FLUENCE_STAGE" = "" ]; then
```
Comment (nahsi) [inline subgraph/fluence-graph.sh:14]:
```suggestion
if [[ -z $BASIC_AUTH_FLUENCE_STAGE ]]; then
```

File: subgraph/fluence-graph.sh [hunk 4]
```diff
@@ -0,0 +1,35 @@
+#!/bin/sh
```
Comment (nahsi) [inline subgraph/fluence-graph.sh:1]:
```suggestion
#! /bin/env bash
```

File: subgraph/fluence-graph.sh [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+# Choose auth string.
+basic_auth=""
+if [ "$BASIC_AUTH_FLUENCE_STAGE" != "" ]; then
+  basic_auth=${BASIC_AUTH_FLUENCE_STAGE}@
+fi
+
+SUBGRAPH_NAME=fluence-deal-contracts
+
+if [ "$1" = "deploy" ]; then
```
Comment (nahsi) [inline subgraph/fluence-graph.sh:27]:
```suggestion
if [[ $1 = "deploy" ]]; then
```

File: subgraph/fluence-graph.sh [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+  echo "Deploy subgraph on Fluence for stage env with subgraph name: $SUBGRAPH_NAME..."
+  graph deploy --node https://${basic_auth}graph-node-admin.fluence.dev/ --ipfs https://${basic_auth}graph-node-ipfs.fluence.dev --network stage --network-file config/networks.json --version-label 0.0.0 ${SUBGRAPH_NAME}
+elif [ "$1" = "create" ]; then
```
Comment (nahsi) [inline subgraph/fluence-graph.sh:30]:
```suggestion
elif [[ "$1" = "create" ]]; then
```

File: subgraph/fluence-graph.sh (hunk 5 above)
Comment (nahsi) [inline subgraph/fluence-graph.sh:27]: I personaly would use `case` here.
```
case $1 in
  deploy)
    echo ... ;;
  create)
    echo ... ;;
  *)
     echo error
     exit 1
  ;;
esac
```

But it is a matter of taste

**Review by nahsi — APPROVED** [review]

File: subgraph/fluence-graph.sh (hunk 2 above)
Comment (AlcibiadesCleinias) [reply subgraph/fluence-graph.sh:21]: even for strings?

**Pull request merged**